docs/content/docs/concepts/Breaking Changes and Supported Patterns.md

```diff
@@ -0,0 +1,293 @@
+3. Make the existing arguments more flexible.
+   For example if the function takes one string, you can replace that argument with a string or configuration object that contains the string and extra optional arguments.
+
+## Module name collisions
+
+User Code which could be broken by a member being added to a module in the Package should not be supported.
+This specifically means User Code that does things like:
+
+```typescript
+export * from "the-package";
+export { nameThatCouldCollide };
+```
+
+Is not supported.
+
+## Imports for other than the root
+
+User code which imports any path not at the root of the package is not supported.
+This allows refactoring the package's internals.
+
+## Undocumented behavior
+
+User code which depends on any behavior of the package which is undocumented should be unsupported.
+
+## Broken behavior
+
+User code which can not work due to a bug in The Package is not Supported.
+This makes changes to the package that cause APIs to not perform as documented incompatible, and fixing such issues compatible.
+
+For example if a function in the package takes in a `number | string`, and always fails to behave as documented when given a string,
+changing the type to `number` is an Incomparable Change since it could cause code that sometimes worked (or even always worked) to stop building.
+However changing the implementation to produce a better error in this case, or updating the documentation to reflect the actual behavior is a compatible change.
+
+This mostly means that APIs that do not work as documented can have their documentation changed to reflect the previously incorrectly documented behavior and/or the runtime behavior in those cases can be changed (usually fixed or the quality of error improved).
+
+## Globals
+
+The Package should not use globals (note that this means actual globals, like members on `globalThis` not module scope).
+This allows user code to use and depend on packages which do use globals without risk of collisions.
+Any exceptions to this need to be documented.
+
+## Enums
+
+The Package should specify if enums it declares can have more items added later or not.
+This can be seen as a special case of Version Type Variance defined above.
+
+The [non_exhaustive attribute in Rust](https://doc.rust-lang.org/reference/attributes/type_system.html#the-non_exhaustive-attribute) is an example of this.
+It has actual compiler support, but it can also just be a convention (and possible a linter rule to help).
+
+# Related work
+
+[semver-ts](https://www.semver-ts.org/) addresses a very similar topic, but focuses on compile errors and not the more general "issues" of which compile errors are a subset. It covers how this relates to variance and its section on "Avoiding user constructibility" suggests several approaches including the approach of doing so through documentation, which is what the proposed "@out" tag above does.
```

Comment:
I found this after writing this document. I really need to read it more as it covers a lot of what this does, and we might be able to unify with that effort somewhat.